### PR TITLE
[BITETRAC-185] using javascript accessible cookie for access token

### PR DESCRIPTION
**Pull request by BlueEyedHush**

### CODE REVIEW

File: src/node_modules/server/auth/local/index.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ router.post('/', function(req, res, next) {
         /* 19.01.2038 03:14:07 GMT - seems to be maximum, cross-browser compatibile expiration time
          * http://stackoverflow.com/a/34586818/3182262 */
         res.cookie('access_token', token,
-          {httpOnly: true, expires: new Date(2038, 1, 19, 3, 14, 7)/* @ToDo: secure: true*/});
+          {httpOnly: false, expires: new Date(2038, 1, 19, 3, 14, 7)});
```
Comment (quirk0o) [inline src/node_modules/server/auth/local/index.js]: Is this secure?

Comment (BlueEyedHush) [inline src/node_modules/server/auth/local/index.js]: No, it's not. But simplifies logout logic significantly - with httponly cookie server must be contacted in order to perform logout. To make logout possible in offline mode you'd have to introduce secondary cookie. 
https://jira.iisg.agh.edu.pl/browse/PPBITETRAC-198

Comment (quirk0o) [inline src/node_modules/server/auth/local/index.js]: Why do we need to make logout possible offline?

Comment (BlueEyedHush) [inline src/node_modules/server/auth/local/index.js]: Not being able to feels unnatural - how can we handle it gracefully? Grey out button when offline? Pretend that you've logged out, but after refreshing you'll still be logged in?

Comment (quirk0o) [inline src/node_modules/server/auth/local/index.js]: Maybe we shouldn't redirect based on the presence of a cookie but some other application state that says whether the user is logged in. So clicking the button would simply toggle that state and queue some kind of request to the server

Comment (BlueEyedHush) [inline src/node_modules/server/auth/local/index.js]: Yes, it's going to be that way - but after merging index & app.

**Pull request merged**